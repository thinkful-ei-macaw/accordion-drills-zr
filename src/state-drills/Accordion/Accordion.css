* {
  box-sizing: border-box;
  text-align: center;
  list-style: none;
  margin: auto;
}

ul, #root{
  width: 90%;
}

ul, li {
  padding: 10px;
  border: 3px solid black;
  margin: 10px 0 10px 0;
}

button {
  padding: 10px;
}