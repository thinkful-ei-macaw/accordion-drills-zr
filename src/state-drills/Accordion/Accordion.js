import React, { Component } from 'react';

class Accordion extends Component {

  static defaultProps = {
    sections: [{
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },]

  }

  handleSetActiveSection(section) {
    this.setState({ activeSection: section })
  }


  state = {
    activeSection: null

  };



  renderItem(idX, activeSection, section) {
    return (
      <li className="AccordionItem" key={idX}>
        <button onClick={() => this.handleSetActiveSection(idX)}>{section.title}</button>
        {(activeSection === idX) && <p>{section.content}</p>}
      </li>
    );
  }


  render() {
    const { activeSection } = this.state
    const { sections } = this.props

    return (
      <ul className="Accordion">
        {sections.map((section, idX) =>
          this.renderItem(idX, activeSection, section)
        )}
      </ul>
    )
  }

}

export default Accordion