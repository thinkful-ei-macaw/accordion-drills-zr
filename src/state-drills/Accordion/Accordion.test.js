import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

const sectionsProp = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe('Accordion', () => {
  it('renders empty without Accordion', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('renders no sections by default', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('opens clicked section', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('only opens one section', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})