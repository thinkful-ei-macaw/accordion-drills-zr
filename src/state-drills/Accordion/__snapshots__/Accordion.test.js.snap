// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion only opens one section 1`] = `
<ul
  className="Accordion"
>
  <li
    className="AccordionItem"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="AccordionItem"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="AccordionItem"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion opens clicked section 1`] = `
<ul
  className="Accordion"
>
  <li
    className="AccordionItem"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="AccordionItem"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="AccordionItem"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion renders empty without Accordion 1`] = `
<ul
  className="Accordion"
>
  <li
    className="AccordionItem"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="AccordionItem"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="AccordionItem"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion renders no sections by default 1`] = `
<ul
  className="Accordion"
>
  <li
    className="AccordionItem"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="AccordionItem"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="AccordionItem"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
